lib/charms/operator_libs_linux/v1/systemd.py

```diff
@@ -66,149 +71,146 @@
 class SystemdError(Exception):
     """Custom exception for SystemD related errors."""
 
-    pass
 
+def _systemctl(*args: str, check: bool = False) -> Optional[int]:
+    """Control a system service using systemctl.
 
-def _popen_kwargs():
-    return {
-        "stdout": subprocess.PIPE,
-        "stderr": subprocess.STDOUT,
-        "bufsize": 1,
-        "universal_newlines": True,
-        "encoding": "utf-8",
-    }
+    Args:
+        *args: Arguments to pass to systemctl.
+        check:
+            Check the output of the systemctl command. Default: False.
+
+    Raises:
+        SystemdError:
+            Raised if returncode != 0 and check is True.
+            None will be returned instead of a returncode.
+    """
+    cmd = ["systemctl", *args]
+    logger.debug(f"Executing command: {cmd}")
+    try:
+        proc = subprocess.run(
+            [*cmd],
```

Comment:
I think `[*cmd]` is equivalent to just `cmd`.

Reply:
Refactored in commit https://github.com/canonical/operator-libs-linux/pull/106/commits/c951f68b8fe18cbab6fc69858e838283063c1e48